### ISSUE TITLE
“Cannot read property 'lastIndexOf' of undefined” when passing filename to spectral lint

### ISSUE THREAD

**Issue by martinbean** [stoplightio/spectral#1097]
**Describe the bug**
When specifying a path (like I have always done) with `spectral lint`, it throws the following error:

> Cannot read property 'lastIndexOf' of undefined

Even with verbose mode, no other information (such as file or line number) is printed.

However, if I run the same command with just a path (i.e. leaving the filename out), it works as intended.

**To Reproduce**

1. Run, i.e. `spectral lint reference/vsapi/openapi.yaml`
2. See error above printed in console
3. Run `spectral lint reference/vsapi`
4. See `No results with a severity of 'error' or higher found!` printed in console.

**Expected behavior**
The help text for `spectral lint` says:

```
Positionals:
  documents  Location of JSON/YAML documents. Can be either a file, a glob or fetchable resource(s) on the web.
```

But, it seems specifying a file breaks it (or at least it does in my case). Am I still supposed to be able to use a file path in Spectral v5? As the error doesn’t happen in Spectral v4.

**Environment:**
 - Library version: 5.3.0
 - OS: macOS Catalina (10.15.4)

**Comment by r1ckr** [comment]
If this helps, I had version 5.1.0 installed on my mac and it linted the definition OK. 
Then saw the API definition CI failing which pulls the latest 5.3.0 (I've learned my lesson). 
Upgraded my mac to 5.3.0 to troubleshoot and now I see the same error.
Downgraded my mac to 5.1.0 and the error persists.

Looks like a dependency that wasn't locked.

One more thing, linting doesn't work when specifying the path as @martinbean says, the output is indeed `No results with a severity of 'error' or higher found!` but even when breaking something in my definition still shows the same message.

**Comment by r1ckr** [comment]
More info

If you run with brackets it doesn't show the error:
```
$ spectral lint [reference/my-api.v1.yml]
No results with a severity of 'error' or higher found!
```

But I've introduced big errors into `my-api.v1.yml`, and it is showing that all is good.

**Comment by P0lip** [comment]
Could you provide the spec you run Spectral against?

**Comment by laat** [comment]
The stacktrace that has this message:
```
TypeError: Cannot read property 'lastIndexOf' of undefined
    at Object.defineProperty (/Users/laat/git/documentation/node_modules/@stoplight/ordered-object-literal/src/index.cjs:18:24)
    at Function.freeze (<anonymous>)
    at deepFreeze (/Users/laat/git/documentation/node_modules/immer/dist/immer.js:104:10)
    at Immer.maybeFreeze (/Users/laat/git/documentation/node_modules/immer/dist/immer.js:1092:17)
    at Immer.finalize (/Users/laat/git/documentation/node_modules/immer/dist/immer.js:978:10)
    at finalizeProperty (/Users/laat/git/documentation/node_modules/immer/dist/immer.js:1053:22)
    at /Users/laat/git/documentation/node_modules/immer/dist/immer.js:75:52
    at Array.forEach (<anonymous>)
    at each (/Users/laat/git/documentation/node_modules/immer/dist/immer.js:75:20)
    at Immer.finalizeTree (/Users/laat/git/documentation/node_modules/immer/dist/immer.js:1084:3)
```

To reproduce:
```
wget https://raw.githubusercontent.com/stoplightio/spectral/develop/src/__tests__/__fixtures__/petstore.oas3.json
npx @stoplight/spectral lint petstore.oas3.json
```

**Comment by mort** [comment]
Hi, just wanted to confirm I'm seeing the same problems reported in this thread (OSX 10.15.3 / Spectral 5.3.0)

- `spectral lint config/service_apis/custom/mcr.yml` fails with `Cannot read property 'lastIndexOf' of undefined`
- `spectral lint config/service_apis/custom/*` apparently succeeds with `No results with a severity of 'error' or higher found!` 
- `spectral lint [config/service_apis/custom/mcr.yml]` also results in `No results with a severity of 'error' or higher found!` 
- `touch config/service_apis/custom/foo.yml && spectral lint [config/service_apis/custom/foo.yml]` also results in `No results with a severity of 'error' or higher found!`

**Comment by r1ckr** [comment]
I've created this repo for you:
https://github.com/r1ckr/spectral-test

I am using this one:
https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/examples/v3.0/petstore.yaml
Changed line `13:20` to have the `operationId` as **snake_case**
Added a `.spectral.yml` to throw that as an error

Error we are describing here:
```
$ spectral lint reference/petstore.yaml
Cannot read property 'lastIndexOf' of undefined
```

Skipping the error, but then linting doesn't work:
```
$ spectral lint [reference/petstore.yaml]
No results with a severity of 'error' or higher found!
```

**Comment by fidgi** [comment]
Hi,
I've the same issue on Linux Debian 4.9.210-1 w/ spectral 5.x.x launched with 'npx @stoplight/spectral@5.x.x lint reference/openapi.yaml'

It works with spectral version 4.2.0

**Comment by nicolastakashi** [comment]
The same issue on Ubuntu 18.04

**Comment by r1ckr** [comment]
Hey @fidgi does the 4.2.0 catches lint errors?

**Comment by fidgi** [comment]
@r1ckr it looks like since the following warning is reported : 
 1:1  warning  api-servers  OpenAPI `servers` must be present and non-empty array.

✖ 1 problem (0 errors, 1 warning, 0 infos, 0 hints)

**Comment by mjisaak** [comment]
Same issue here. I have an Azure DevOps build pipeline that acts as a gate. Since 5.3 I get the `Cannot read property 'lastIndexOf' of undefined` error. Downgrading to 5.2 doesn't solve the issue.

**Comment by nulltoken** [comment]
I can repro the issue.

```
MINGW64 /h/spectral ((3a181ce...))
$ npx @stoplight/spectral@latest lint https://raw.githubusercontent.com/stoplightio/spectral/develop/src/__tests__/__fixtures__/petstore.oas3.json
Cannot read property 'lastIndexOf' of undefined
```
However, it doesn't seem to reproduce against the development branch.

```
MINGW64 /h/spectral ((3a181ce...))
$ yarn cli lint https://raw.githubusercontent.com/stoplightio/spectral/develop/src/__tests__/__fixtures__/petstore.oas3.json
yarn run v1.21.1
$ node -r ts-node/register -r tsconfig-paths/register src/cli/index.ts lint https://raw.githubusercontent.com/stoplightio/spectral/develop/src/__te
sts__/__fixtures__/petstore.oas3.json
Missing baseUrl in compilerOptions. tsconfig-paths will be skipped
OpenAPI 3.x detected

https://raw.githubusercontent.com/stoplightio/spectral/develop/src/__tests__/__fixtures__/petstore.oas3.json
  1:1   warning  openapi-tags           OpenAPI object should have non-empty `tags` array.
  3:10  warning  info-contact           Info object should contain `contact` object.
  3:10  warning  info-description       OpenAPI object info `description` must be present and non-empty string.
 17:13  warning  operation-description  Operation `description` must be present and non-empty string.
 20:18  warning  operation-tag-defined  Operation tags should be defined in global tags.
 64:14  warning  operation-description  Operation `description` must be present and non-empty string.
 67:18  warning  operation-tag-defined  Operation tags should be defined in global tags.
 86:13  warning  operation-description  Operation `description` must be present and non-empty string.
 89:18  warning  operation-tag-defined  Operation tags should be defined in global tags.

✖ 9 problems (0 errors, 9 warnings, 0 infos, 0 hints)

Done in 14.59s.
```

A new beta release should be issued shortly by @P0lip. We'll ping you as soon as it's available in order to check with you if the issue is still happening in your context.

**Comment by P0lip** [comment]
My apologies folks for causing the issue. It was me who broke it :sweat_smile: 
For anyone interested here is the PR https://github.com/stoplightio/spectral/pull/1105 with a more in-depth explanation of the issue.
Once it's merged, I'll release beta.

I'll try to add some sort of circuit-breaker in `ordered-object-literal` package later today, so that it does not throw TypeError - doing so could potentially fix `5.3.0` if that newer version is picked up by Npm or Yarn when you install Spectral.

**Comment by nulltoken** [comment]
> More info
> 
> If you run with brackets it doesn't show the error:
> 
> ```
> $ spectral lint [reference/my-api.v1.yml]
> No results with a severity of 'error' or higher found!
> ```
> 
> But I've introduced big errors into `my-api.v1.yml`, and it is showing that all is good.

@r1ckr Sorry to ask, but where does this bracket based syntax come from?

**Comment by P0lip** [comment]
Although, 5.4.0-beta1 has just been released, 5.3.0 should be working again, so feel free to stick to 5.3.0 in case you don't want to try out beta (that's pretty stable anyway, but just called beta)
Just make sure to clean yarn/npm cache and re-install it again.

**Comment by K3UL** [comment]
Hi just to confirm the following (reinstalling 5.3.0) fixed it for me :
```bash
npm remove -g @stoplight/spectral
npm cache clean --force
npm install -g @stoplight/spectral
```
